src/loader.rs

```diff
@@ -209,17 +215,20 @@ impl<'a> LoaderInternal<'a> {
             })
         })() {
             let map = arena_mmap.alloc(map);
-            object = object::File::parse(&**map)?;
-        }
+            Some(object::File::parse(&**map)?)
+        } else {
+            None
+        };
+        let object1 = dsym.as_ref().unwrap_or(&object);
```

Comment:
```suggestion
        let dwarf_object = dsym.as_ref().unwrap_or(&object);
```
This looks correct now, but this might be a clearer name.

Reply:
Resolved in 376f95ac1f26ba097b9c47e64fe602c1eeec3116